### DISCUSSION TITLE
Server side rendering _next/data returns 404 (NGINX)

### DISCUSSION THREAD

**Discussion by tpetersons** [vercel/next.js#32009] [Help] [answered]
Hello everyone,

I've NEXTJS application that's hosted locally with NGINX and PM2.
Once I implemented server side rendering using GetServerSideProps, I've noticing 404 errors for _next/data JSON pages.
When I build project and run production on localhost, it works fine. Only when It's deployed using nginx.
Also I can see that data is appended into html, and page is working, just that json file is unaccessible and returns 404.

Any ideas?
I feel like this might be my nginx config, since its working fine without it.
Here is the config I am using:

```
server {
    server_name diyfpv.lv;
    access_log /opt/diyfpv/logs/access.log;
    error_log /opt/diyfpv/logs/error.log error;

    location /_next/ {
        alias /var/repo/diyfpv/.next/;
        expires 30d;
        access_log on;
    }

    location / {
        # reverse proxy for next server
        proxy_pass http://localhost:5000;
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection 'upgrade';
        proxy_set_header Host $host;
        proxy_cache_bypass $http_upgrade;

        # we need to remove this 404 handling
        # because next's _next folder and own handling
        # try_files $uri $uri/ =404;
    }
}
```

**Comment by wwwdepot** [comment, marked as answer]
`_next/`  contains dynamic content especially when you are using SSR... You can't serve them as static files.

Remove this block & proxy_pass everything to next.js.
```
    location /_next/ {
        alias /var/repo/diyfpv/.next/;
        expires 30d;
        access_log on;
    }
```

**Reply by tpetersons** [reply]
Hi @wwwdepot this makes so much sense now.
Thanks, I have removed this block and sure enough it's all good now!
I had the right idea but I have to learn more about **nginx**.
Have a great day everyone! this is closed.

**Reply by faceless49** [reply]
@tpetersons, @wwwdepot  hello guys. I have the same problem, but my devops says there are no such lines in nginx. I don't know what to do anymore
